src/ServiceInsight/SequenceDiagram/ModelCreator.cs

```diff
@@ -98,6 +99,19 @@ void Initialize()
 
                 sendingHandler.Out = sendingHandler.Out.Concat(new[] { arrow }).OrderBy(a => a).ToList();
             }
+
+            ReOrderHandlers(firstOrderHandlers);
+        }
+
+        private void ReOrderHandlers(IList<Handler> orderedHandlers)
```

Comment:
I disagree with this assessment.

The method is badly named, and could take an `IEnumerable` instead of an `IList`, which would make it more clear that it's assigning the handlers field with the values from the parameter. I don't think that's a mixed responsibility.

I'm also against inlining it. `Initialize` is long enough. It makes the code easier to understand, so there's the justification for it's existence.